zookeeper/source/zk_dn_client.cc

```diff
@@ -0,0 +1,124 @@
+#ifndef RDFS_ZK_CLIENT_DN_CC
+#define RDFS_ZK_CLIENT_DN_CC
+
+#include "zk_dn_client.h"
+#include "zk_lock.h"
+#include <easylogging++.h>
+
+namespace zkclient{
+
+
+    const std::string ZkClientDn::CLASS_NAME = ": **ZkClientDn** : ";
+
+	ZkClientDn::ZkClientDn(const std::string& ip, const std::string& hostname, const std::string& zkIpAndAddress,
+            const uint32_t ipcPort, const uint32_t xferPort) : ZkClientCommon(zkIpAndAddress) {
+
+        data_node_id = DataNodeId();
+        data_node_id.ip = ip;
+        data_node_id.ipcPort = ipcPort;
+
+        // TODO: Fill in other data_node stats
+        data_node_payload = DataNodePayload();
```

Comment:
Is there some way we could save the "hostname" as a vararg or something inside of this payload?